### PR TITLE
Modify Course Edit and Course Achieve

### PR DESCRIPTION
**Pull request by teriaiw**

### CODE REVIEW

File: src/main/java/igrad/commons/core/Messages.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
     public static final String MESSAGE_UNKNOWN_COURSE_COMMAND = "Unknown course command, possible options;\n"
         + "course add ...\n"
         + "course edit ...\n"
+        + "course assign ...\n"
```
Comment (nathanaelseen) [inline src/main/java/igrad/commons/core/Messages.java]: no Teri, this should not be there :)

Comment (nathanaelseen) [reply src/main/java/igrad/commons/core/Messages.java]: its only for `requirement assign`

Comment (teriaiw) [reply src/main/java/igrad/commons/core/Messages.java]: oops, I wrote wrongly, wanted to write Achieve

Comment (nathanaelseen) [reply src/main/java/igrad/commons/core/Messages.java]: oh ok

File: src/main/java/igrad/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
     public static final Prefix PREFIX_DESCRIPTION = new Prefix("d/");
     public static final Prefix PREFIX_SEMESTER = new Prefix("s/");
     public static final Prefix PREFIX_TAG = new Prefix("x/");
+    public static final Prefix PREFIX_CAP = new Prefix("c/");
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/parser/CliSyntax.java:18]: ok

File: src/main/java/igrad/logic/commands/course/CourseEditCommand.java [hunk 3]
```diff
@@ -56,22 +61,36 @@ public CommandResult execute(Model model) throws CommandException {
...
+
+        CourseInfo editedCourse = createEditedCourseInfo(courseToEdit, editCourseDescriptor);
 
-        return null;
+        model.setCourseInfo(editedCourse);
+        return new CommandResult(String.format(MESSAGE_SUCCESS, editedCourse));
     }
 
     @Override
     public boolean equals(Object other) {
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/commands/course/CourseEditCommand.java:79]: good

File: src/main/java/igrad/logic/parser/course/CourseAchieveCommandParser.java [hunk 4]
```diff
@@ -18,7 +27,26 @@
...
+        if (argMultimap.getValue(PREFIX_CAP).isEmpty()) {
+            throw new ParseException(MESSAGE_NOT_CALCULATED);
+        }
+
+        if (argMultimap.getValue(PREFIX_SEMESTER).isEmpty()) {
+            throw new ParseException(MESSAGE_SEMS_LEFT_NEEDED);
+        }
+
+        Cap cap = new Cap(argMultimap.getValue(PREFIX_CAP).get());
+        int semsLeft = Integer.parseInt(argMultimap.getValue(PREFIX_SEMESTER).get());
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/parser/course/CourseAchieveCommandParser.java:47]: good job @teriaiw , i'm still thinking if i should put sems left in course info

File: src/main/java/igrad/model/course/CourseInfo.java [hunk 5]
```diff
@@ -47,12 +47,63 @@ public CourseInfo(Optional<Name> name, Optional<Cap> cap) {
      * requirement list.
      */
     public static Cap computeCap(List<Module> moduleList) {
-        double cap = 4.5;
-        /**
-         * TODO: Teri, please fill the details. Iterate through all modules in the module list and
-         * for each module with a grade, add them together to calcualte the cap
-         * Lemme know if you find this too hard, i can do it :)
-         */
+        double cap = 0;
```
Comment (nathanaelseen) [inline src/main/java/igrad/model/course/CourseInfo.java:50]: good but can be improved; package all this logic into the `model/modules/Grade.java` class

**Review by nathanaelseen — APPROVED** [review]

**Pull request merged**